base/MAPL_newCFIO.F90

```diff
@@ -645,6 +646,37 @@ subroutine stage2DLatLon(this, fileName, oClients, rc)
         ref = ArrayReference(this%lats)
          call oClients%collective_stage_data(this%write_collection_id,trim(filename),'lats', &
               ref,start=localStart, global_start=GlobalStart, global_count=GlobalCount)
+        deallocate(LocalStart,GlobalStart,GlobalCount)
+     end if
+
+     var_lon => this%metadata%get_variable('corner_lons')
+     var_lat => this%metadata%get_variable('corner_lats')
+     
+     hasll = associated(var_lon) .and. associated(var_lat)
+     if (hasll) then
+        factory => get_factory(this%output_grid,rc=status)
+        _VERIFY(status)
+
+        call factory%generate_file_edge_bounds(this%output_grid,LocalStart,GlobalStart,GlobalCount,rc=status)
+        _VERIFY(status)
+        call ESMF_GridGetCoord(this%output_grid, localDE=0, coordDim=1, &
+        staggerloc=ESMF_STAGGERLOC_CORNER, &
+        farrayPtr=ptr2d, rc=status)
+        _VERIFY(STATUS)
+        if (.not.allocated(this%corner_lons)) allocate(this%corner_lons(size(ptr2d,1),size(ptr2d,2)))
+        this%corner_lons=ptr2d*MAPL_RADIANS_TO_DEGREES
```

Comment:
I think you can do this more elegantly.  No need for the explicit allocate.   Just
```
this%corner_lons = ptr2d * MAPL_RADIANS_TO_DEGREES
```

Reply:
Is '=' overloaded to do the allocation and match the shape/bounds of the right hand side if the left hand side is not allocated?

Reply:
"Overloaded" is the wrong term here.  In F2003, _intrinsic_ assignment with allocatable LHS is required to allocate (or reallocated) the LHS unless it already matches the shape of the RHS.   Exactly suited to this case.

F2008 went a step further and included reallocating if the LHS is polymorphic and not of the same type as the rhs.

Reply:
Make the requested changes including an updating the mockgridfactory so hopefully the MAPL unit tests pass this time.